Source/WebKit/UIProcess/Extensions/WebExtensionStorageSQLiteStore.cpp

```diff
@@ -0,0 +1,428 @@
+        }
+
+        String errorMessage;
+        auto results = keys.size() ? protectedThis->getValuesForKeysWithErrorMessage(keys, errorMessage) : protectedThis->getValuesForAllKeys(errorMessage);
+
+        WorkQueue::protectedMain()->dispatch([results = WTFMove(results), errorMessage = WTFMove(errorMessage), completionHandler = WTFMove(completionHandler)]() mutable {
+            completionHandler(results, errorMessage);
+        });
+    });
+}
+
+static size_t storageSizeOf(HashMap<String, String> map)
+{
+    size_t totalStorage = 0;
+
+    for (auto key : map.keys())
+        totalStorage += key.sizeInBytes();
+
+    for (auto value : map.values())
+        totalStorage += value.sizeInBytes();
+
+    return totalStorage;
+}
+
+void WebExtensionStorageSQLiteStore::getStorageSizeForKeys(Vector<String> keys, CompletionHandler<void(size_t storageSize, const String& errorMessage)>&& completionHandler)
+{
+    queue()->dispatch([protectedThis = Ref { *this }, uniqueIdentifier = uniqueIdentifier(), keys = WTFMove(keys), completionHandler = WTFMove(completionHandler)]() mutable {
+        String errorMessage;
+        if (!keys.isEmpty()) {
+            auto keysAndValues = protectedThis->getValuesForKeysWithErrorMessage(keys, errorMessage);
+            WorkQueue::protectedMain()->dispatch([keysAndValues = WTFMove(keysAndValues), errorMessage = WTFMove(errorMessage), completionHandler = WTFMove(completionHandler)]() mutable {
+                completionHandler(storageSizeOf(keysAndValues), errorMessage);
+            });
+
+            return;
+        }
+
+        // Return storage size for all keys if no keys are specified.
+        if (!protectedThis->openDatabaseIfNecessary(errorMessage, false)) {
+            WorkQueue::protectedMain()->dispatch([errorMessage = WTFMove(errorMessage), completionHandler = WTFMove(completionHandler)]() mutable {
+                completionHandler(0, errorMessage);
+            });
+            return;
+        }
+
+        int64_t result = 0;
+        RefPtr<API::Error> error;
+        bool success = SQLiteDatabaseEnumerate(*(protectedThis->database()), error, "SELECT SUM(LENGTH(key) + LENGTH(value)) FROM extension_storage"_s, std::tie(result));
+
+        WorkQueue::protectedMain()->dispatch([success, result, error = WTFMove(error), protectedThis = Ref { protectedThis }, keys = WTFMove(keys), completionHandler = WTFMove(completionHandler)]() mutable {
+            if (success)
+                completionHandler(result, { });
+            else {
+                RELEASE_LOG_ERROR(Extensions, "Failed to calculate storage size for keys: %s for extension %s. %s", rowFilterStringFromRowKeys(keys).utf8().data(), protectedThis->uniqueIdentifier().utf8().data(), error->localizedDescription().utf8().data());
+                completionHandler(0, error->localizedDescription());
+            }
+        });
+    });
+}
+
+static Vector<String> hashmapToVector(HashMap<String, String> map, bool mapKeys)
```

Comment:
No need to put the param type in the name.
```suggestion
static Vector<String> toVector(HashMap<String, String> map, bool mapKeys)
```